ext/url/00_url.js

```diff
@@ -363,10 +399,18 @@
       }
     }
 
+    #hasAuthority() {
+      // https://github.com/servo/rust-url/blob/1d307ae51a28fecc630ecec03380788bfb03a643/url/src/lib.rs#L824
+      return this.#serialization.slice(this.#schemeEnd).startsWith("://");
+    }
+
     /** @return {string} */
     get hash() {
       webidl.assertBranded(this, URLPrototype);
-      return this[_url].hash;
+      // https://github.com/servo/rust-url/blob/1d307ae51a28fecc630ecec03380788bfb03a643/url/src/quirks.rs#L263
+      return this.#fragmentStart
+        ? trim(this.#serialization.slice(this.#fragmentStart))
+        : "";
```

Comment:
I wonder if we should lazy cache this (applies to all part getters).

Reply:
sure we can, not this PR though...already too much complexity - I also think we should do that global LRU cache first.